front/src/reducers/osrdconf/helpers.ts

```diff
@@ -142,3 +145,47 @@ export const updateDestinationPathStep = (
   destination: Partial<PathStep> | null,
   replaceCompletely: boolean = false
 ) => updatePathStepAtIndex(pathSteps, pathSteps.length - 1, destination, replaceCompletely);
+
+/**
+ * modifies the array statePathSteps in place
+ */
+export function upsertPathStep(statePathSteps: (PathStep | null)[], op: SuggestedOP) {
+  // We know that, at this point, origin and destination are defined because pathfinding has been done
+  const cleanPathSteps = compact(statePathSteps);
+
+  let newVia = {
+    ...pick(op, [
+      'coordinates',
+      'positionOnPath',
+      'name',
+      'ch',
+      'kp',
+      'stopFor',
+      'arrival',
+      'locked',
+      'deleted',
+      'onStopSignal',
+      'theoreticalMargin',
+    ]),
+    id: nextId(),
+    ...(op.uic
+      ? { uic: op.uic }
+      : {
+          track: op.track,
+          offset: op.offsetOnTrack,
+        }),
+  } as PathStep;
```

Comment:
Maybe it's better not to force the type

```suggestion
  let newVia: PathStep = {
    ...pick(op, [
      'coordinates',
      'positionOnPath',
      'name',
      'ch',
      'kp',
      'stopFor',
      'arrival',
      'locked',
      'deleted',
      'onStopSignal',
      'theoreticalMargin',
    ]),
    id: nextId(),
    ...(op.uic
      ? { uic: op.uic }
      : {
          track: op.track,
          offset: op.offsetOnTrack,
        }),
  };
```

Reply:
done